samples/BikeSharingApp/BikeSharingWeb/server.js

```diff
@@ -67,5 +67,9 @@ function getApiUrl(host) {
     var apiName = process.env.API_NAME || "gateway";
 
     // return full URL of API service (spacePrefix + rootSpace + apiName + host)
-    return prefix + hostArr[root] + "." + apiName + "." + baseHost;
+    if (prefix !== "")
+    {
+        return prefix + hostArr[root] + "." + apiName + "." + baseHost;
+    }
+    return host.replace("bikesharingweb", apiName);
```

Comment:
>return host.replace("bikesharingweb", apiName); [](start = 4, length = 47)

I'm a bit confused by this one. I understand what it does, but I don't understand why it cannot be used in the case where prefix !== "". Even if we have a prefix, that should work right? Or is this just to handle the case where the prefix used is also "bikesharingweb" ([http://bikesharingweb.bikesharingweb.1234.nip.io](http://bikesharingweb.bikesharingweb.1234.nip.io))?

Reply:
yes the host.replace should work if prefix is not null as well. I think the reason why we constructed the URL in a more complicated way before was that we could construct the URL without the string "bikesharingweb".
Unfortunately we can't do that anymore. I'll go ahead and pass bikesharingweb as an environment variable with a default value if not found and I'll return host.replace("bikesharingweb", apiName) for all cases. Let me know if that works.